src/Cli/func/Helpers/GlobalCoreToolsSettings.cs

```diff
@@ -99,6 +99,13 @@ public static void Init(ISecretsManager secretsManager, string[] args)
             {
                 _currentWorkerRuntime = WorkerRuntime.None;
             }
+
+            if (_currentWorkerRuntime == WorkerRuntime.Java || args.Any(arg => string.Equals(arg, WorkerRuntime.Java.ToString(), StringComparison.OrdinalIgnoreCase)))
```

Comment:
The second check is necessary. Without it, commands that have `java` but do not have `--java` in it (such as `func new --language java` will be sent down this path and fail:

```
                else
                {
                    _currentWorkerRuntime = WorkerRuntimeLanguageHelper.GetCurrentWorkerRuntimeLanguage(secretsManager);
                }
```